### PR TITLE
Bug jsx build

### PR DESCRIPTION
**Pull request by msecret**
The karma warnings plugin is great at ensuring syntax errors in tests
don't stop the test file from being tested at all, but creates really
bad error messages during the build. This fix would make that plugin
only run on CI, when we really need to know if tests aren't being run
due to syntax errors.

Ref #743

### CODE REVIEW

**Comment by msecret** [comment]
I tested this by purposely committing a syntax error. Circle did stop the build as expected, but I still was able to see accurate errors locally:

https://circleci.com/gh/18F/cg-dashboard/556

**Comment by adborden** [comment]
Why do we even need the `WebpackKarmaWarningsPlugin`? If we leave it out, will we also get better message formatting in circle?

**Comment by msecret** [comment]
@adborden if we leave it out, and there's a syntax error in a test file, it will continue running the tests, skipping the broken file completely.

**Comment by adborden** [comment]
Hmm, I see. Instead of hooking into `NODE_ENV==production` for tests which seems weird to me, can we create a new condition around `NODE_ENV==test`? That's a more common convention I've seen.

**Comment by adborden** [comment]
Or if it's really specific to Circle/CI we can use the `CI` env variable https://circleci.com/docs/environment-variables/#basics

**Comment by msecret** [comment]
I changed it to use "test"

File: circle.yml [hunk 1]
```diff
@@ -37,7 +37,9 @@ test:
     - if ! go get github.com/golang/tools/cmd/cover; then go get golang.org/x/tools/cmd/cover; fi
   override:
     - cd $WS && ./codecheck.sh -u
-    - cd $WS && npm test
+    - cd $WS && npm test:
+        environment:
+          NODE_ENV: "test"
```
Comment (adborden) [inline circle.yml]: yaml fanciness, how does this look as json? Mind if we write this instead:

```yaml
- cd $WS && NODE_ENV=test npm test
```

**Comment by adborden** [comment]
Quoting Marco Segreto (2016-11-23 10:38:41)
> 
> Sure, I was hoping I could apply the environment to the whole `test` block, but couldn't anyway, so that's fine.

Hmm, yeah that would be nice. One option would be creating a sh script, or
an npm script.

--
Aaron D Borden
Human and Hacker

**Pull request merged**